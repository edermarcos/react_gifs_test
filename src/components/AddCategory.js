import React from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategory }) => {
  const [inputValue, setInputValue] = React.useState('');

  const handleSubmit = (e) => {
    e.preventDefault();

    if (!inputValue.trim()) {
      return;
    }

    /*
     * Se recibe como parametro el setter desde el padre
     * Por default, entites es el parametro que retorna y se crea un nuevo arreglo
     * */
    setCategory(entities => [inputValue, ...entities]);
    setInputValue('');
  };

  return (
    <form onSubmit={ handleSubmit }>
      <p>{ inputValue }</p>
      <input type="text"
             value={ inputValue }
             onChange={ (event) => setInputValue(event.target.value) }/>
    </form>
  );
};

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
};
