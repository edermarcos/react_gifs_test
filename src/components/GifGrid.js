import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category, deleteCategoryById }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <div>
      <div className="header">
        <h3>Category: { category }</h3>

        <div>
          <button onClick={ () => deleteCategoryById(category) }>Delete</button>
        </div>
      </div>

      { loading && <p>Loading...</p> }

      <div className="card-grid">
        {
          images.map(entity => (
            <GifGridItem key={ entity.id }
                         { ...entity }
            />
          ))
        }
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
  deleteCategoryById: PropTypes.func,
};
