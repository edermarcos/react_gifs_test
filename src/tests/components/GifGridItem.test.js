import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Tests for <GifGridItem />', () => {
  const title = 'This a title';
  const url = 'https://images.pexels.com/photos/9824398/pexels-photo-9824398.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500';
  const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);

  test('Should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show a paragraph with the value of the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Should display the image correctly', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  // Should have the class from animate.css library
  test('Should have the class from animate.css library', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
