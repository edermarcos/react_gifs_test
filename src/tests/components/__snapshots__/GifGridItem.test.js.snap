// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGridItem /> Should display the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="This a title"
    src="https://images.pexels.com/photos/9824398/pexels-photo-9824398.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
  />
  <p>
    This a title
  </p>
</div>
`;
